apps/browser/app/page.tsx

```diff
@@ -1,56 +1,158 @@
 "use client";
 
-import Chat from "@/components/Chat";
-import { evoServiceAtom, chatIdAtom } from "@/lib/store";
+import {
+  evoServiceAtom,
+  workspaceAtom,
+  workspaceFilesAtom,
+  chatIdAtom,
+  localOpenAiApiKeyAtom,
+  showAccountModalAtom,
+  errorAtom,
+  newGoalSubmittedAtom,
+  isChatLoadingAtom,
+} from "@/lib/store";
+import { useCreateChat } from "@/lib/mutations/useCreateChat";
+import { useUpdateChatTitle } from "@/lib/mutations/useUpdateChatTitle";
 import { EvoService } from "@/lib/services/evo/EvoService";
+import { useEvoService } from "@/lib/hooks/useEvoService";
+import { useWorkspaceUploadUpdate } from "@/lib/hooks/useWorkspaceUploadUpdate";
+import Chat from "@/components/Chat";
 import { useEffect, useState } from "react";
 import { useSession } from "next-auth/react";
 import { useRouter } from "next/navigation";
 import { useAtom } from "jotai";
+import { v4 as uuid } from "uuid";
+import { InMemoryFile } from "@nerfzael/memory-fs";
 
 function Dojo({ params }: { params: { id?: string } }) {
+  const [evoService, setEvoService] = useAtom(evoServiceAtom);
+  const [chatId, setChatId] = useAtom(chatIdAtom);
+  const [newGoalSubmitted, setNewGoalSubmitted] = useAtom(newGoalSubmittedAtom);
```

Comment:
Since this component rerenders fully upon the params.id changing, it gets hard reset if it is kept within useState. This was a bug I found, but could very well have been caused by something else, but fairly confident this is needed. I'm not an expert on how next.js handles client-side page reroutes.

Reply:
Ah I see. It's because when `router` pushing, so that `params.id` changes you are actually navigating pages. You go from your current page to another that does contain an id parameter; hence the page unmounts (along with its local state) and then re-mounts. So you persist it in the global state so that it persists across pages.

I wonder if this is better than having 2 pages; some logic might be duplicated but it'd be more legible/easier to track these things.

Reply:
Ah I see so the problem is going from the empty "/" to a filled in "chat/<id>", which causes the state to be wiped. Is that correct?

Yes I think having a page that doesn't do this would be ideal to avoid future problems like this, very hard to debug.